### PR TITLE
Add missed test cases for sycl::vec

### PR DESCRIPTION
**Pull request by kopylovanat**
Added missed test cases for:

- unary operators -,+
- operator%
- operator DataT()
- noexcept check for size() and byte_size()
- check for swizzled vec for operator[]
- constructor vec(const ArgTN&... args)
- element_type

### CODE REVIEW

File: tests/vector_constructors/generate_vector_constructors.py [hunk 1]
```diff
@@ -117,6 +145,25 @@ def generate_vec(type_str, size):
...
+    input_vec_size = size//2
+    values_size = size - input_vec_size
+    vals_list1 = append_fp_postfix(type_str, Data.vals_list_dict[size][:input_vec_size])
+    vals_list2 = append_fp_postfix(type_str, Data.vals_list_dict[size][-values_size:])
+    test_string = mixed_constructor_vec_template.substitute(
+        type=type_str,
+        size=size,
+        input_vec_size=input_vec_size,
+        vals1=', '.join(vals_list1),
+        vals2=', '.join(vals_list2))
```
Comment (keryell) [inline tests/vector_constructors/generate_vector_constructors.py:161]:
```suggestion
        type = type_str,
        size = size,
        input_vec_size = input_vec_size,
        vals1 = ', '.join(vals_list1),
        vals2 = ', '.join(vals_list2))
```

File: tests/vector_api/generate_vector_api.py [hunk 2]
```diff
@@ -96,6 +107,10 @@ def gen_checks(type_str, size):
         type_str, kernel_name,
         'API test for sycl::vec<' + type_str + ', ' + str(size) + '>',
         test_string)
+    string+= vector_element_type_template.substitute(
+        type=type_str,
+        size=size,
+        swizIndexes=', '.join(Data.swizzle_elem_list_dict[size][::-1]))
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply tests/vector_api/generate_vector_api.py]: Added space to `string +=` but for substitute parameters it is formatted like this without spaces everywhere in python generation files. Do you think it's better to update this formatting everywhere?

[the remark this answers is not in the document]
Comment (keryell) [reply tests/vector_api/generate_vector_api.py]: Ah yes you are right. Follow the current style.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**